transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -1100,7 +1096,11 @@ boolean setAddComplete() {
         // Ensure we never update when the handlerState is REMOVE_COMPLETE already.
         // oldState is usually ADD_PENDING but can also be REMOVE_COMPLETE when an EventExecutor is used that is not
         // exposing ordering guarantees.
-        return HANDLER_STATE_UPDATER.getAndSet(this, ADD_COMPLETE) != REMOVE_COMPLETE;
```

Comment:
The existing logic here looks incorrect, shouldn't it be a conditional update, i.e. not set it back to `ADD_COMPLETE` if it's already `REMOVE_COMPLETE`? The new logic looks better though!

Reply:
Yep... I have an upcoming PR fo 4.1 as well. Stay tuned :)

Reply:
Actually it is already correct in `4.1` :)